src/data/mystery-encounters/mystery-encounter.ts

```diff
@@ -0,0 +1,963 @@
+import { EncounterPokemonRequirement, EncounterSceneRequirement, HealthRatioRequirement, PartySizeRequirement, StatusEffectRequirement, WaveRangeRequirement } from "./mystery-encounter-requirements";
+import { BattlerIndex } from "#app/battle";
+import { EncounterAnim } from "#app/data/battle-anims";
+import { MysteryEncounterTier } from "#enums/mystery-encounter-tier";
+import { MysteryEncounterMode } from "#enums/mystery-encounter-mode";
+import { MysteryEncounterOptionMode } from "#enums/mystery-encounter-option-mode";
+import { GameModes } from "#app/game-mode";
+
+export interface EncounterStartOfBattleEffect {
+  sourcePokemon?: Pokemon;
+  sourceBattlerIndex?: BattlerIndex;
+  targets: BattlerIndex[];
+  move: PokemonMove;
+  ignorePp: boolean;
+  followUp?: boolean;
+}
+
+/**
+ * Used by {@linkcode MysteryEncounterBuilder} class to define required/optional properties on the {@linkcode MysteryEncounter} class when building.
+ *
+ * Should ONLY contain properties that are necessary for {@linkcode MysteryEncounter} construction.
+ * Post-construct and flag data properties are defined in the {@linkcode MysteryEncounter} class itself.
+ */
+export interface IMysteryEncounter {
+  encounterType: MysteryEncounterType;
+  options: [MysteryEncounterOption, MysteryEncounterOption, ...MysteryEncounterOption[]];
+  spriteConfigs: MysteryEncounterSpriteConfig[];
+  encounterTier: MysteryEncounterTier;
+  encounterAnimations?: EncounterAnim[];
+  disabledGameModes?: GameModes[];
+  hideBattleIntroMessage: boolean;
+  autoHideIntroVisuals: boolean;
+  enterIntroVisualsFromRight: boolean;
+  catchAllowed: boolean;
+  continuousEncounter: boolean;
+  maxAllowedEncounters: number;
+  hasBattleAnimationsWithoutTargets: boolean;
+  skipEnemyBattleTurns: boolean;
+  skipToFightInput: boolean;
+
+  onInit?: (scene: BattleScene) => boolean;
+  onVisualsStart?: (scene: BattleScene) => boolean;
+  doEncounterExp?: (scene: BattleScene) => boolean;
+  doEncounterRewards?: (scene: BattleScene) => boolean;
+  doContinueEncounter?: (scene: BattleScene) => Promise<void>;
+
+  requirements: EncounterSceneRequirement[];
+  primaryPokemonRequirements: EncounterPokemonRequirement[];
+  secondaryPokemonRequirements: EncounterPokemonRequirement[];
+  excludePrimaryFromSupportRequirements: boolean;
+
+  dialogue: MysteryEncounterDialogue;
+  enemyPartyConfigs: EnemyPartyConfig[];
+
+  dialogueTokens: Record<string, string>;
+  expMultiplier: number;
+}
+
+/**
+ * MysteryEncounter class that defines the logic for a single encounter
+ * These objects will be saved as part of session data any time the player is on a floor with an encounter
+ * Unless you know what you're doing, you should use MysteryEncounterBuilder to create an instance for this class
+ */
+export default class MysteryEncounter implements IMysteryEncounter {
+  /**
+   * Required params
+   */
```

Comment:
you could use 
```suggestion
  // #region Optional params
  
```

This would show up like this on vs-codium minimap:
<img width="133" alt="image" src="https://github.com/user-attachments/assets/7b7e86c9-c384-4e34-ae51-162c2097b936">